src/Microsoft.AspNetCore.WebHooks.Receivers/IWebHookRequestReader.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Microsoft.AspNetCore.WebHooks
+{
+    /// <summary>
+    /// Provides an abstraction for a service that parses the request body. For use in filters that execute prior to
+    /// regular model binding or in actions that read the request body after regular model binding.
+    /// </summary>
+    public interface IWebHookRequestReader
```

Comment:
Consider making this an abstract class. It seems like it might grow. I'm pretty careful now -- if an interface has more than one use case, an abstract class is usually going to work better in the end.

Reply:
@rynowak and I discussed this point offline. Agreed `IWebHookRequestReader` is fine as-is.